core/src/main/java/com/linecorp/armeria/common/HttpRequest.java

```diff
@@ -279,24 +292,48 @@ static HttpRequest of(RequestHeaders headers, Publisher<? extends HttpObject> pu
      * >     }
      * > }
      * }</pre>
+     *
+     * @see #withHeaders(RequestHeaders)
      */
-    static HttpRequest of(HttpRequest request, RequestHeaders newHeaders) {
+    static HttpRequest of(RequestHeaders newHeaders, HttpRequest request) {
```

Comment:
Can we just remove this method and make users to use `withHeaders` directly?

Reply:
Hmm. Maybe that's OK.